src/Components/Components/src/NavigationManagerExtensions.cs

```diff
@@ -0,0 +1,694 @@
+        /// added or updated.
+        /// </summary>
+        /// <param name="navigationManager">The <see cref="NavigationManager"/>.</param>
+        /// <param name="name">The name of the paramter to add or update.</param>
+        /// <param name="value">The value of the parameter to add or update.</param>
+        public static string UriWithQueryParameter(this NavigationManager navigationManager, string name, long value)
+            => UriWithQueryParameter(navigationManager, name, Format(value));
+
+        /// <summary>
+        /// Returns a <see cref="string"/> equal to <see cref="NavigationManager.Uri"/> except with a single parameter
+        /// added, updated, or removed.
+        /// </summary>
+        /// <param name="navigationManager">The <see cref="NavigationManager"/>.</param>
+        /// <param name="name">The name of the paramter to add or update.</param>
+        /// <param name="value">The value of the parameter to add or update.</param>
+        /// <remarks>
+        /// If <paramref name="value"/> is <c>null</c>, the parameter will be removed if it exists in the URI.
+        /// Otherwise, it will be added or updated.
+        /// </remarks>
+        public static string UriWithQueryParameter(this NavigationManager navigationManager, string name, long? value)
+            => UriWithQueryParameter(navigationManager, name, Format(value));
+
+        /// <summary>
+        /// Returns a <see cref="string"/> equal to <see cref="NavigationManager.Uri"/> except with a single parameter
+        /// added, updated, or removed.
+        /// </summary>
+        /// <param name="navigationManager">The <see cref="NavigationManager"/>.</param>
+        /// <param name="name">The name of the paramter to add or update.</param>
+        /// <param name="value">The value of the parameter to add or update.</param>
+        /// <remarks>
+        /// If <paramref name="value"/> is <c>null</c>, the parameter will be removed if it exists in the URI.
+        /// Otherwise, it will be added or updated.
+        /// </remarks>
+        public static string UriWithQueryParameter(this NavigationManager navigationManager, string name, string? value)
+        {
+            if (navigationManager is null)
+            {
+                throw new ArgumentNullException(nameof(navigationManager));
+            }
+
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(EmptyQueryParameterExceptionMessage, nameof(name));
+            }
+
+            var uri = navigationManager.Uri;
+
+            return value is null
+                ? UriWithoutQueryParameter(uri, name)
+                : UriWithQueryParameterCore(uri, name, value);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="string"/> equal to <see cref="NavigationManager.Uri"/> except with a single parameter
```

Comment:
What does _single parameter_ mean?

Reply:
I would say something along the lines of

> Returns a URI that is constructed by updating `NavigationManager.Uri` with a ..

Reply:
On the next line:
> added or updated.

I'll apply your wording though!